File: DSA/Vaishnavi_590017838/Day_12/Question1.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
        int lines = 1;
        int currentWidth = 0;
        for(int i = 0; i < s.length(); i++) {
            int letterWidth = widths[s[i] - 'a'];
            if(currentWidth + letterWidth > 100) {
                lines++;
                currentWidth = letterWidth;
            }
            else {
                currentWidth += letterWidth;
            }
        }
        return {lines, currentWidth};
    }
};
int main() {
    vector<int> widths(26,10);
    string s = "abcdefghijklmnopqrstuvwxyz";
    Solution obj;
    vector<int> ans = obj.numberOfLines(widths,s);
    cout << ans[0] << " " << ans[1];

    return 0;
}

File: DSA/ShlokMadan_590017851/Day_12/Question1.java
import java.util.Arrays;

public class Question1 {

    public static int[] numberOfLines(int[] widths, String s) {

        int lines = 1;
        int currentWidth = 0;

        for (int i = 0; i < s.length(); i++) {

            int width = widths[s.charAt(i) - 'a'];

            if (currentWidth + width > 100) {
                lines++;
                currentWidth = width;
            } else {
                currentWidth += width;
            }
        }

        return new int[]{lines, currentWidth};
    }

    public static void main(String[] args) {

        int[] widths = new int[26];
        Arrays.fill(widths, 10);

        String s = "abcdefghijklmnopqrstuvwxyz";

        int[] ans = numberOfLines(widths, s);

        System.out.println(Arrays.toString(ans));
    }
}